spec.md

```diff
@@ -513,6 +522,41 @@ message PluginCapability {
 
 If the plugin is unable to complete the GetPluginCapabilities call successfully, it MUST return a non-ok gRPC code in the gRPC status.
 
+#### `Probe`
+
+A Plugin MUST implement this RPC call.
+The primary utility of the Probe RPC is deployment verification: this can be a one time or periodic check to ensure that a plugin instance is healthy and has all dependencies available.
+This information can be used, for example, to monitor the health of the plugin and redeploy the plugin (or take other automated measures) when it becomes unhealthy.
```

Comment:
Deployment verification and checking of dependencies is something I would lump in to "initialization checks" and not the responsibility of `Probe`. By implying that it is, we implicitly expect a `Probe` call to happen around deployment time, and it may not happen until sometime afterwards.

I would suggest rewording this to be as generic as possible: `The primary utility of the Probe RPC is to verify the plugin is in a healthy state. If an unhealthy state is reported, via a non-success response, the CO MAY take action (such as restarting the plugin container) to try and bring the plugin to a healthy state.`